packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -0,0 +1,186 @@
+          '.vscode-insiders',
+      fs.path.join(homeDirPath, 'Applications',
+          'Visual Studio Code - Insiders.app', 'Contents'): '.vscode-insiders'
+    };
+
+    return _findInstalled(stable, insiders);
+  }
+
+  // Windows:
+  //   $programfiles(x86)\Microsoft VS Code
+  //   $programfiles(x86)\Microsoft VS Code Insiders
+  // TODO: Confirm these are correct for 64bit
+  //   $programfiles\Microsoft VS Code
+  //   $programfiles\Microsoft VS Code Insiders
+  // Windows Extensions:
+  //   $HOME/.vscode/extensions
+  //   $HOME/.vscode-insiders/extensions
+  static List<VsCode> _installedWindows() {
+    final String progFiles86 = platform.environment['programfiles(x86)'];
+    final String progFiles = platform.environment['programfiles'];
+
+    final Map<String, String> stable = <String, String>{
+      fs.path.join(progFiles86, 'Microsoft VS Code'): '.vscode',
+      fs.path.join(progFiles, 'Microsoft VS Code'): '.vscode'
+    };
+    final Map<String, String> insiders = <String, String>{
+      fs.path.join(progFiles86, 'Microsoft VS Code Insiders'):
+          '.vscode-insiders',
+      fs.path.join(progFiles, 'Microsoft VS Code Insiders'): '.vscode-insiders'
+    };
+
+    return _findInstalled(stable, insiders);
+  }
+
+  // Linux:
+  //   /usr/share/code/bin/code
+  //   /usr/share/code-insiders/bin/code-insiders
+  // Linux Extensions:
+  //   $HOME/.vscode/extensions
+  //   $HOME/.vscode-insiders/extensions
+  static List<VsCode> _installedLinux() {
+    return _findInstalled(
+      <String, String>{'/usr/share/code': '.vscode'},
+      <String, String>{'/usr/share/code-insiders': '.vscode-insiders'}
+    );
+  }
+
+  static List<VsCode> _findInstalled(
+      Map<String, String> stable, Map<String, String> insiders) {
+    final Map<String, String> allPaths = <String, String>{};
+    allPaths.addAll(stable);
+    if (_includeInsiders)
+      allPaths.addAll(insiders);
+
+    final List<VsCode> results = <VsCode>[];
+
+    for (String directory in allPaths.keys) {
+      if (fs.directory(directory).existsSync()) {
+        final String extensionDirectory =
+            fs.path.join(homeDirPath, allPaths[directory], 'extensions');
+        results.add(new VsCode.fromDirectory(directory, extensionDirectory));
+      }
+    }
+
+    return results;
+  }
+
+  void _init() {
+    _isValid = false;
+    _validationMessages.clear();
+
+    if (!fs.isDirectorySync(directory)) {
+      _validationMessages.add('VS Code not found at $directory');
+      return;
+    }
+
+    // If the extensions directory doesn't exist at all, the listSync()
+    // below will fail, so just bail out early.
+    if (!fs.isDirectorySync(extensionDirectory)) {
+      return;
+    }
+
+    // Check for presence of extension.
+    final Iterable<FileSystemEntity> extensionDirs = fs
+        .directory(extensionDirectory)
+        .listSync()
+        .where((FileSystemEntity d) => fs.isDirectorySync(d.path))
```

Comment:
`d is Directory`

Reply:
Aha, I was sure there was a better way to do this, but was looking at the methods/properties on the instance. Thanks! (Changed in 4f8ebf4).